src/benchmark.cc

```diff
@@ -463,10 +488,18 @@ void RunBenchmarks(const std::vector<Benchmark::Instance>& benchmarks,
     flushStreams(display_reporter);
     flushStreams(file_reporter);
     for (const auto& benchmark : benchmarks) {
-      std::vector<BenchmarkReporter::Run> reports =
-          RunBenchmark(benchmark, &complexity_reports);
-      display_reporter->ReportRuns(reports);
-      if (file_reporter) file_reporter->ReportRuns(reports);
+      bool display_aggregates_only;
+      std::pair<std::vector<BenchmarkReporter::Run> /*everything*/,
+                std::vector<BenchmarkReporter::Run> /*aggregates_only*/>
+          reports = RunBenchmark(benchmark, &complexity_reports,
+                                 display_aggregates_only);
+
+      const std::vector<BenchmarkReporter::Run>& report_for_display =
+          display_aggregates_only ? reports.second /*aggregates_only*/
```

Comment:
would this also be simpler if the return type was 'non-aggregate', 'aggregate' instead of having the first be everything?

Reply:
Hmm, actually it might be.
I didn't do so initially because i thought it was important to print the whole benchmark at once
(for header/context pre-print), but maybe it is too pessimistic..